### ISSUE TITLE
Invalid hotSpot

### ISSUE THREAD

**Issue by stefan-reich** [TigerVNC/tigervnc#843]
**Describe the bug**
The viewer says "Invalid hotSpot" and crashes.

**To Reproduce**

Connect from Linux client to Linux server with SSH tunneling using the TigerVNC Java client.

**Expected behavior**
Normal stuff

**Client (please complete the following information):**
 - TigerVNC Java 1.9.0

**Log**

```
Tunnel: Next authentication method: publickey
Tunnel: Authentication succeeded (publickey).
CConn: connected to host localhost port 5901
CConnection: Server supports RFB protocol version 3.8
CConnection: Using RFB protocol version 3.8
Exception in thread "AWT-EventQueue-0" java.lang.ClassCastException: java.desktop/javax.swing.JLayer cannot be cast to java.desktop/javax.swing.AbstractButton
	at java.desktop/javax.swing.plaf.synth.SynthButtonUI.getComponentState(SynthButtonUI.java:154)
	at java.desktop/javax.swing.plaf.synth.SynthButtonUI.getContext(SynthButtonUI.java:135)
	at java.desktop/javax.swing.plaf.synth.SynthBorder.paintBorder(SynthBorder.java:56)
	at java.desktop/javax.swing.JComponent.paintBorder(JComponent.java:967)
	at java.desktop/javax.swing.JComponent.paint(JComponent.java:1075)
	at java.desktop/javax.swing.JLayer.paint(JLayer.java:475)
	at java.desktop/javax.swing.plaf.LayerUI.paint(LayerUI.java:80)
	at java.desktop/javax.swing.plaf.ComponentUI.update(ComponentUI.java:161)
	at java.desktop/javax.swing.JComponent.paintComponent(JComponent.java:797)
	at java.desktop/javax.swing.JLayer.paint(JLayer.java:470)
	at java.desktop/javax.swing.JComponent.paintChildren(JComponent.java:907)
	at java.desktop/javax.swing.JComponent.paint(JComponent.java:1083)
	at java.desktop/javax.swing.JComponent.paintChildren(JComponent.java:907)
	at java.desktop/javax.swing.JComponent.paint(JComponent.java:1083)
	at java.desktop/javax.swing.JComponent.paintChildren(JComponent.java:907)
	at java.desktop/javax.swing.JComponent.paint(JComponent.java:1083)
	at java.desktop/javax.swing.JComponent.paintChildren(JComponent.java:907)
	at java.desktop/javax.swing.JComponent.paint(JComponent.java:1083)
	at java.desktop/javax.swing.JComponent.paintChildren(JComponent.java:907)
	at java.desktop/javax.swing.JComponent.paint(JComponent.java:1083)
	at java.desktop/javax.swing.JComponent.paintChildren(JComponent.java:907)
	at java.desktop/javax.swing.JComponent.paint(JComponent.java:1083)
	at java.desktop/javax.swing.JComponent.paintChildren(JComponent.java:907)
	at java.desktop/javax.swing.JComponent.paint(JComponent.java:1083)
	at java.desktop/javax.swing.JLayeredPane.paint(JLayeredPane.java:590)
	at java.desktop/javax.swing.JComponent.paintChildren(JComponent.java:907)
	at java.desktop/javax.swing.JComponent.paintToOffscreen(JComponent.java:5262)
	at java.desktop/javax.swing.BufferStrategyPaintManager.paint(BufferStrategyPaintManager.java:246)
	at java.desktop/javax.swing.RepaintManager.paint(RepaintManager.java:1313)
	at java.desktop/javax.swing.JComponent.paint(JComponent.java:1060)
	at java.desktop/java.awt.GraphicsCallback$PaintCallback.run(GraphicsCallback.java:39)
	at java.desktop/sun.awt.SunGraphicsCallback.runOneComponent(SunGraphicsCallback.java:78)
	at java.desktop/sun.awt.SunGraphicsCallback.runComponents(SunGraphicsCallback.java:115)
	at java.desktop/java.awt.Container.paint(Container.java:2000)
	at java.desktop/java.awt.Window.paint(Window.java:3940)
	at java.desktop/javax.swing.RepaintManager$4.run(RepaintManager.java:868)
	at java.desktop/javax.swing.RepaintManager$4.run(RepaintManager.java:840)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:87)
	at java.desktop/javax.swing.RepaintManager.paintDirtyRegions(RepaintManager.java:840)
	at java.desktop/javax.swing.RepaintManager.paintDirtyRegions(RepaintManager.java:815)
	at java.desktop/javax.swing.RepaintManager.prePaintDirtyRegions(RepaintManager.java:764)
	at java.desktop/javax.swing.RepaintManager.access$1200(RepaintManager.java:69)
	at java.desktop/javax.swing.RepaintManager$ProcessingRunnable.run(RepaintManager.java:1880)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:313)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:770)
	at java.desktop/java.awt.EventQueue.access$600(EventQueue.java:97)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:715)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:87)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:740)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:117)
	at java.desktop/java.awt.WaitDispatchSupport$2.run(WaitDispatchSupport.java:190)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:313)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:770)
	at java.desktop/java.awt.EventQueue.access$600(EventQueue.java:97)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:715)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:87)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:740)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
Exception in thread "AWT-EventQueue-0" java.lang.ClassCastException: java.desktop/javax.swing.JLayer cannot be cast to java.desktop/javax.swing.AbstractButton
	at java.desktop/javax.swing.plaf.synth.SynthButtonUI.getComponentState(SynthButtonUI.java:154)
	at java.desktop/javax.swing.plaf.synth.SynthButtonUI.getContext(SynthButtonUI.java:135)
	at java.desktop/javax.swing.plaf.synth.SynthBorder.paintBorder(SynthBorder.java:56)
	at java.desktop/javax.swing.JComponent.paintBorder(JComponent.java:967)
	at java.desktop/javax.swing.JComponent.paint(JComponent.java:1075)
	at java.desktop/javax.swing.JLayer.paint(JLayer.java:475)
	at java.desktop/javax.swing.plaf.LayerUI.paint(LayerUI.java:80)
	at java.desktop/javax.swing.plaf.ComponentUI.update(ComponentUI.java:161)
	at java.desktop/javax.swing.JComponent.paintComponent(JComponent.java:797)
	at java.desktop/javax.swing.JLayer.paint(JLayer.java:470)
	at java.desktop/javax.swing.JComponent.paintChildren(JComponent.java:907)
	at java.desktop/javax.swing.JComponent.paint(JComponent.java:1083)
	at java.desktop/javax.swing.JComponent.paintToOffscreen(JComponent.java:5255)
	at java.desktop/javax.swing.BufferStrategyPaintManager.paint(BufferStrategyPaintManager.java:246)
	at java.desktop/javax.swing.RepaintManager.paint(RepaintManager.java:1313)
	at java.desktop/javax.swing.JComponent._paintImmediately(JComponent.java:5203)
	at java.desktop/javax.swing.JComponent.paintImmediately(JComponent.java:5013)
	at java.desktop/javax.swing.JLayer.paintImmediately(JLayer.java:430)
	at java.desktop/javax.swing.plaf.LayerUI.paintImmediately(LayerUI.java:723)
	at java.desktop/javax.swing.JLayer.paintImmediately(JLayer.java:425)
	at java.desktop/javax.swing.JComponent.paintImmediately(JComponent.java:4994)
	at java.desktop/javax.swing.RepaintManager$4.run(RepaintManager.java:857)
	at java.desktop/javax.swing.RepaintManager$4.run(RepaintManager.java:840)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:87)
	at java.desktop/javax.swing.RepaintManager.paintDirtyRegions(RepaintManager.java:840)
	at java.desktop/javax.swing.RepaintManager.paintDirtyRegions(RepaintManager.java:815)
	at java.desktop/javax.swing.RepaintManager.prePaintDirtyRegions(RepaintManager.java:764)
	at java.desktop/javax.swing.RepaintManager.access$1200(RepaintManager.java:69)
	at java.desktop/javax.swing.RepaintManager$ProcessingRunnable.run(RepaintManager.java:1880)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:313)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:770)
	at java.desktop/java.awt.EventQueue.access$600(EventQueue.java:97)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:715)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:87)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:740)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:117)
	at java.desktop/java.awt.WaitDispatchSupport$2.run(WaitDispatchSupport.java:190)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:313)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:770)
	at java.desktop/java.awt.EventQueue.access$600(EventQueue.java:97)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:715)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:87)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:740)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
CConn: Using pixel format depth 24 (32bpp) little-endian rgb888
CConn: Using Tight encoding
java.lang.IndexOutOfBoundsException: invalid hotSpot
	at java.desktop/sun.awt.CustomCursor.<init>(CustomCursor.java:80)
	at java.desktop/sun.awt.X11CustomCursor.<init>(X11CustomCursor.java:44)
	at java.desktop/sun.awt.X11.XCustomCursor.<init>(XCustomCursor.java:43)
	at java.desktop/sun.awt.X11.XToolkit.createCustomCursor(XToolkit.java:1297)
	at com.tigervnc.vncviewer.Viewport.setCursor(Viewport.java:221)
	at com.tigervnc.vncviewer.Viewport.setCursor(Viewport.java:211)
	at com.tigervnc.vncviewer.DesktopWindow.setCursor(DesktopWindow.java:287)
	at com.tigervnc.vncviewer.CConn.setCursor(CConn.java:385)
	at com.tigervnc.rfb.CMsgReader.readSetXCursor(CMsgReader.java:266)
	at com.tigervnc.rfb.CMsgReader.readMsg(CMsgReader.java:99)
	at com.tigervnc.rfb.CConnection.processMsg(CConnection.java:136)
	at com.tigervnc.vncviewer.VncViewer.run(VncViewer.java:565)
	at java.base/java.lang.Thread.run(Thread.java:844)

```

**Comment by bphinz** [comment]
I'm sorry for taking so long to get to this.  I can't seem to reproduce the issue, can you tell me a little bit more about the environment, etc.?  Is there anything specific that allows the issue to be triggered consistently?

**Comment by samhed** [comment]
No response

**Comment by patrickSpaceSurfer** [comment]
This happens to me. Server is Xvnc on a FreeBSD machine (tightvnc-1.3.10_6) , client is running on arch linux (tigervnc 1.12.0-3, jdk-openjdk 18-2). Connection is over the internet via an ssh tunnel. It happens consistently when I use a scaling factor above 100% in either full screen or window mode. Note that the non java vncviewer works flawlessly (but without scaling). Here is the full exception:

java -jar /usr/share/vnc/classes/VncViewer.jar

TigerVNC Java Viewer v1.12.0 (20220314)
Built on 2022-03-14 at 12:48:11
Copyright (C) 1999-2021 TigerVNC Team and many others (see README.rst)
See https://www.tigervnc.org for information on TigerVNC.
DecodeManager: Detected 8 CPU core(s)
DecodeManager: Creating 4 decoder thread(s)
CConn: connected to host localhost port 5901
CConnection: Server supports RFB protocol version 3.8
CConnection: Using RFB protocol version 3.8
CConn: Using pixel format depth 24 (32bpp) little-endian rgb888
java.lang.IndexOutOfBoundsException: invalid hotSpot
        at java.desktop/sun.awt.CustomCursor.<init>(CustomCursor.java:80)
        at java.desktop/sun.awt.X11CustomCursor.<init>(X11CustomCursor.java:44)
        at java.desktop/sun.awt.X11.XCustomCursor.<init>(XCustomCursor.java:43)
        at java.desktop/sun.awt.X11.XToolkit.createCustomCursor(XToolkit.java:1171)
        at com.tigervnc.vncviewer.Viewport.setCursor(Viewport.java:221)
        at com.tigervnc.vncviewer.Viewport.setCursor(Viewport.java:211)
        at com.tigervnc.vncviewer.DesktopWindow.setCursor(DesktopWindow.java:281)
        at com.tigervnc.vncviewer.CConn.setCursor(CConn.java:336)
        at com.tigervnc.rfb.CMsgReader.readSetXCursor(CMsgReader.java:271)
        at com.tigervnc.rfb.CMsgReader.readMsg(CMsgReader.java:99)
        at com.tigervnc.rfb.CConnection.processMsg(CConnection.java:149)
        at com.tigervnc.vncviewer.VncViewer.run(VncViewer.java:430)
        at java.base/java.lang.Thread.run(Thread.java:833)

**Comment by fnJeff** [comment]
Unfortunately, I don't think this is fixed in java client 1.14.0. I just tried this version for the first time today and experienced the same crash described in https://github.com/TigerVNC/tigervnc/issues/843. Everything seemed fine until I clicked in a konsole and typed 1 character. Viewer immediately disappeared and the Invalid hotspot error popup window showed up. I don't know exactly what info you need to help debug, but if you let me know, I will try and get you what I can.
As described above, it definitely is related to the resizing option (the whole reason I still use the java client).
